snap7/util/db.py

```diff
@@ -137,6 +137,50 @@
 logger = getLogger(__name__)
 
 
+def prepare_tia_export_to_parse(txt_path: str) -> str:
+    """Return a string that can be ingested by parse_specification
+    from a .txt file directly copied and pasted from TIA Portal.
+    It also handles duplicate variable names by progressively appending “_X”.
+
+    Args:
+        tia_export: path to the .txt
+
+    Returns:
+        string ready to be parsed
+    """
+
+    with open(txt_path, "r") as file:
+        db_specification = ""
+
+        valid_list = ["BOOL", "DWORD", "INT", "DINT", "CHAR", "STRING", "DATE_AND_TIME", "TIME_OF_DAY", "REAL", "BYTE"]
+        var_names: list[str] = []
+
+        for line in file:
+            line = line.lstrip("\t")
+            parsed_line = line.split("\t")
+
+            var_name = parsed_line[0]
+            var_type = parsed_line[1].upper()
+            var_offset = parsed_line[2]
+
+            to_add = "_0"
+            for name in reversed(var_names):
+                name = str(name)
+                if name.rsplit("_")[0] == var_name:
+                    print(name.rsplit("_")[-1])
+                    to_add = "_" + str(int(name.rsplit("_")[-1]) + 1)
+                    break
+
+            var_name = var_name + to_add
+            var_names.append(var_name)
+
+            if var_type:
+                if var_type in valid_list:
+                    new_line = var_offset + "\t" + var_name + "\t" + var_type
+                    db_specification = db_specification + "\n" + new_line
```

Comment:
nit: Perhaps appending "new_line" to a list and then using join() is more efficient. But I’m not sure it makes a significant difference